disco/extensions/upgrade_simulation/upgrades/voltage_upgrade_functions.py

```diff
@@ -885,36 +924,111 @@ def determine_new_regulator_upgrades(voltage_config, buses_with_violations, volt
+        if remove_commands_list:
+            for x in dss_commands_list:
+                if x in remove_commands_list:
+                    dss_commands_list.remove(x)
         reload_dss_circuit(dss_file_list=dss_file_list, commands_list=dss_commands_list, **kwargs)
         comparison_dict["disabled_new_regcontrol"] = compute_voltage_violation_severity(
                                                         voltage_upper_limit=voltage_upper_limit, voltage_lower_limit=voltage_lower_limit, **kwargs)
 
     return {"new_reg_upgrade_commands": new_reg_upgrade_commands, "comparison_dict": comparison_dict, "best_setting_so_far": best_setting_so_far}
 
 
+def get_newly_added_regulator_settings(dss_file_list, previous_dss_commands_list, regcontrol_cluster_commands, voltage_lower_limit, voltage_upper_limit,
+                                       voltage_config, deciding_field, **kwargs):
+    """this function finalizes the settings with the newly added voltage regulator. 
+    It also takes into account errors encountered due to max control iterations being exceeded.
+    """
+    # reload circuit after settings sweep
+    try:
+        logger.info("Settings sweep for existing reg control devices (other than sub LTC).")
+        regcontrol_df = get_regcontrol_info()
+        regcontrol_sweep_df = sweep_regcontrol_settings(voltage_config=voltage_config,
+                                                        initial_regcontrols_df=regcontrol_df,
+                                                        voltage_upper_limit=voltage_upper_limit, voltage_lower_limit=voltage_lower_limit,
+                                                        exclude_sub_ltc=True, only_sub_ltc=False, **kwargs)
+        regcontrols_df, regcontrol_settings_commands_list = choose_best_regcontrol_sweep_setting(deciding_field=deciding_field,
+            regcontrol_sweep_df=regcontrol_sweep_df, initial_regcontrols_df=regcontrol_df, **kwargs)
+        reload_dss_circuit(dss_file_list=dss_file_list, commands_list=previous_dss_commands_list + regcontrol_cluster_commands + regcontrol_settings_commands_list, **kwargs)  # reload circuit after settings sweep
+        reg_upgrade_commands = regcontrol_cluster_commands + regcontrol_settings_commands_list
+    except Exception as e:  # if there is an error: dss._cffi_api_util.DSSException: (#485)
+        logger.info(f"First attempt at regulator settings sweep failed with error: {e}.")
+        reload_dss_circuit(dss_file_list=dss_file_list, commands_list=previous_dss_commands_list, **kwargs)  # reload circuit before clustering        
+        temp = []
+        for command in regcontrol_cluster_commands:  # extract only new addition commands
+            if "edit regcontrol." not in command.lower():
+                temp.append(command)
+            else:
+                continue        
+        # control iterations are exceeded
+        increase_control_iterations = dss.Solution.MaxControlIterations() + 50

```

Comment:
Should 50 be a config parameter?

Reply:
I think it's fine to leave this as is. This is trying to increase iterations and see if it resolves the maxcontroliterations error